lunatrace/bsl/backend/src/github/projects.ts

```diff
@@ -68,48 +138,93 @@ function getGithubAccessToken(identity: Identity) {
   return githubProvider.initial_access_token;
 }
 
-githubApiRouter.post('/github/user/projects', async (req, res) => {
+export interface KratosIdentityConfig {
+  providers: KratosIdentityProvider[];
+}
+
+export interface KratosIdentityProvider {
+  initial_id_token: string;
+  subject: string;
+  provider: string;
+  initial_access_token: string;
+  initial_refresh_token: string;
+}
+
+githubApiRouter.post('/webhook/github/login', async (req, res) => {
   const kratosConfig = new Configuration({
     basePath: 'http://localhost:4434',
   });
 
   const kratosApiClient = V0alpha2ApiFactory(kratosConfig);
 
-  const userId = req.body.user_id;
+  const userId = req.body.ctx.identity.id;
 
   const identity = await kratosApiClient.adminGetIdentity(userId, ['oidc']);
 
   const accessToken = getGithubAccessToken(identity.data);
 
-  const graphqlWithAuth = graphql.defaults({
-    headers: {
-      authorization: `token ${accessToken}`,
-    },
-  });
+  const github = generateGithubGraphqlClient(accessToken);
 
-  try {
-    const response = await graphqlWithAuth(
-      `query ($login: String!) {
-      organization(login: $login) {
-        repositories(privacy: PRIVATE) {
-          totalCount
-        }
-      }
-    }`,
-      { login: 'breadchris' }
-    );
-    res.send(response);
-  } catch (e) {
-    console.error(e);
-    if (e instanceof GraphqlResponseError) {
-      res.send({
-        error: e.message,
-      });
-    } else {
-      // TODO (cthompson) what other error types exist?
-      res.send({
-        error: e,
-      });
-    }
+  let orgsAfter: string | null | undefined = undefined;
+  let allUserOrgs: GetUserOrganizationsQuery | null = null;
+  let moreDataAvailable = true;
+
+  while (moreDataAvailable) {
+    const userOrgs: GetUserOrganizationsQuery = await github.GetUserOrganizations({
+      orgsAfter: orgsAfter,
+    });
+
+    allUserOrgs = deepmerge(allUserOrgs || {}, userOrgs);
+
+    if (!userOrgs.viewer) break;
+
+    const orgPageInfo = userOrgs.viewer.organizations.pageInfo;
+
+    orgsAfter = orgPageInfo.hasNextPage ? orgPageInfo.startCursor : null;
+    moreDataAvailable = !!orgsAfter;
   }
+
+  console.log('got orgs for user');
+
+  if (allUserOrgs === null || !allUserOrgs.viewer) {
+    throw new Error('github user organizations response is null');
```

Comment:
Nit: If you're going to throw like this, you need to catch it somewhere or else this will dump crap on the consumer of the API.

When you rip this function out to fetch GitHub orgs for a user, I would recommend wrapping your call to that function in a `try/catch` block inside of the Express Controller. If it catches anything then you can return a standard 

```ts
console.error('Failed to fetch GitHub org data for user:', e);
res.status(500).send({error: true, message: 'bad response from GitHub API.');
```